node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -887,6 +910,19 @@ async fn process_incoming_peer_message<Context>(
 
 				peer_data.set_collating(collator_id, para_id);
 			} else {
+				if state.active_paras.is_next(&para_id) {
+					gum::debug!(
+						target: LOG_TARGET,
+						peer_id = ?origin,
+						?collator_id,
+						?para_id,
+						"Declared as collator for the next rotation",
+					);
+
+					peer_data.set_collating(collator_id, para_id);
+					return
```

Comment:
This `return` looks a bit error prone to me. It falls completely out of line, with the rest. The same indentation level in an else would not be too bad, for that simple reputation change anyway.